wagtailio/core/models/pages.py

```diff
@@ -130,3 +138,165 @@ class ContentPage(Page, HeroMixin, SocialMediaMixin, CrossPageMixin):
+
+    content_panels = Page.content_panels + [
+        FieldPanel("introduction"),
+        FieldPanel("cta"),
+        InlinePanel("showcase_items", label="Showcase items"),
+    ]
+
+    promote_panels = (
+        Page.promote_panels + SocialMediaMixin.panels + CrossPageMixin.panels
+    )
+
+    search_fields = Page.search_fields + [
+        index.SearchField("description"),
+        index.SearchField("introduction"),
+    ]
+
+    def _is_htmx_request(self, request):
+        return request.headers.get("HX-Request") == "true"
+
+    def _filter_used_sectors(self):
+        return ["All"] + list(
+            self.showcase_items.values_list("sector__name", flat=True).distinct()
+        )
+
+    def _filtered_showcase_items(self, sector):
+        if sector in self._filter_used_sectors() and sector != "All":
+            return self.showcase_items.filter(sector__name=sector)
+        else:
+            return self.showcase_items.all()
+
+    def get_template(self, request, *args, **kwargs):
+        if self._is_htmx_request(request):
+            return self.htmx_template
+        else:
+            return self.template
+
+    def serve(self, request):
+        response = super().serve(request)
+        if self._is_htmx_request(request):
+            # We only return the fragment, rather than the whole page.
+            new_url = self.url + "?" + request.GET.urlencode()
+            response.headers["HX-Push"] = new_url
+        return response
+
+    def get_context(self, request):
+        context = super().get_context(request)
+        sectors = {}
+
+        for sector in self._filter_used_sectors():
+            if sector == request.GET.get("sector"):
+                sectors[sector] = "selected"
+            else:
+                sectors[sector] = ""
+
+        context |= {
+            "sectors": sectors,
+            "showcase_items": self._filtered_showcase_items(request.GET.get("sector")),
+            "current_sector": request.GET.get("sector") or "All",
+        }
+
+        return context
+
+
+class ShowcaseItem(Orderable):
+    title = models.CharField(max_length=255)
+    description = models.TextField(blank=True)
+    image = models.ForeignKey(
+        "images.WagtailIOImage",
+        models.SET_NULL,
+        null=True,
+        related_name="+",
+    )
+    alt_text = models.CharField(max_length=255, blank=True)
+    logo = models.ForeignKey(
+        "images.WagtailIOImage",
+        models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="+",
+    )
+    external_link = models.URLField(blank=True)
+    internal_link = models.ForeignKey(
+        "wagtailcore.Page",
+        models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name="+",
+    )
+    link_text = models.CharField(max_length=255)
+
+    sector = models.ForeignKey(
+        "Sector",
+        models.CASCADE,
+        null=False,
+        blank=False,
+        related_name="+",
+    )
+
+    page = ParentalKey(
```

Comment:
nitpick: the convention is to have this first. it reduces the cognitive load / scanning time in understanding the setup

Reply:
Sure thing.